DataSourceTransactionManagerFactory is being auto-configured

**Micronaut  version**: 1.0.0

Hi,

I think I have found a possible bug. 

**Bigger context**: 

I have created a redis library using micronaut and I want to use it in a Spring Boot project. I want micronaut to provide `Bean`s (Singleton, Context, etc) that I have configured in `micronaut-redis-library` and that should be it. 

But, micronaut is trying to configure: `io.micronaut.spring.tx.datasource.DataSourceTransactionManagerFactory` and the reason why it’s trying to do is because: both the conditions below are satisfied. 

```
@Factory
@Requires(classes = DataSourceTransactionManager.class)
@Requires(condition = HibernatePresenceCondition.class)
public class DataSourceTransactionManagerFactory {
...
}
```

My `MicronautBeanProcessor` config:

```
@Bean
    public MicronautBeanProcessor beanProcessor() {
        return new MicronautBeanProcessor(Context.class, Singleton.class);
    }
```

Notice that in the stacktrace below Micronaut is trying to create `org.springframework.jdbc.datasource.DataSourceTransactionManager` but there’s no `DataSource` I want configured via `micronaut`. I have `mysql` datasource configured in Spring and I want Spring to answer to that. 

**Recommended Solution**

Do not autoconfigure `DataSourceTransactionManager` if a given property is present:

`io.micronaut.datasource.disabled`


_Stacktrace:_

```
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'org.springframework.jdbc.datasource.DataSourceTransactionManager': FactoryBean threw exception on object creation; nested exception is io.micronaut.context.exceptions.BeanInstantiationException: Could Not Create Bean [class org.springframework.jdbc.datasource.DataSourceTransactionManager]
    at org.springframework.beans.factory.support.FactoryBeanRegistrySupport.doGetObjectFromFactoryBean(FactoryBeanRegistrySupport.java:178)
    at org.springframework.beans.factory.support.FactoryBeanRegistrySupport.getObjectFromFactoryBean(FactoryBeanRegistrySupport.java:101)
    at org.springframework.beans.factory.support.AbstractBeanFactory.getObjectForBeanInstance(AbstractBeanFactory.java:1645)
    at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.getObjectForBeanInstance(AbstractAutowireCapableBeanFactory.java:1182)
    at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:257)
    at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:199)
    at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:251)
    at org.springframework.beans.factory.support.DefaultListableBeanFactory.addCandidateEntry(DefaultListableBeanFactory.java:1325)
    at org.springframework.beans.factory.support.DefaultListableBeanFactory.findAutowireCandidates(DefaultListableBeanFactory.java:1291)
    at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1101)
    at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1065)
    at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:815)
    at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:721)
    ... 64 more
Caused by: io.micronaut.context.exceptions.BeanInstantiationException: Could Not Create Bean [class org.springframework.jdbc.datasource.DataSourceTransactionManager]
    at io.micronaut.spring.beans.MicronautSpringBeanFactory.getObject(MicronautSpringBeanFactory.java:67)
    at org.springframework.beans.factory.support.FactoryBeanRegistrySupport.doGetObjectFromFactoryBean(FactoryBeanRegistrySupport.java:171)
    ... 76 more
```

**Screenshot**

![image](https://user-images.githubusercontent.com/141136/47536564-aa049380-d875-11e8-854c-27fa5c69c891.png)

--- a/configurations/hibernate-jpa/src/main/java/io/micronaut/configuration/hibernate/jpa/HibernateTransactionManagerFactory.java
+++ b/configurations/hibernate-jpa/src/main/java/io/micronaut/configuration/hibernate/jpa/HibernateTransactionManagerFactory.java
@@ -17,11 +17,10 @@
 package io.micronaut.configuration.hibernate.jpa;
 
 import io.micronaut.context.annotation.*;
-import io.micronaut.spring.tx.datasource.DataSourceTransactionManagerFactory;
+import io.micronaut.jdbc.spring.DataSourceTransactionManagerFactory;
 import org.hibernate.SessionFactory;
 import org.springframework.orm.hibernate5.HibernateTransactionManager;
 
-import javax.inject.Singleton;
 import javax.sql.DataSource;
 
 /**

--- a/configurations/jdbc-dbcp/src/test/groovy/io/micronaut/configuration/jdbc/dbcp/DatasourceFactorySpec.groovy
+++ b/configurations/jdbc-dbcp/src/test/groovy/io/micronaut/configuration/jdbc/dbcp/DatasourceFactorySpec.groovy
@@ -16,7 +16,7 @@
 
 package io.micronaut.configuration.jdbc.dbcp
 
-import io.micronaut.spring.tx.datasource.SpringDataSourceResolver
+import io.micronaut.jdbc.spring.SpringDataSourceResolver
 import org.apache.commons.dbcp2.BasicDataSource
 import org.springframework.jdbc.datasource.TransactionAwareDataSourceProxy
 import spock.lang.Specification

--- a/configurations/jdbc-tomcat/src/test/groovy/io/micronaut/configuration/jdbc/tomcat/DatasourceFactorySpec.groovy
+++ b/configurations/jdbc-tomcat/src/test/groovy/io/micronaut/configuration/jdbc/tomcat/DatasourceFactorySpec.groovy
@@ -16,7 +16,7 @@
 
 package io.micronaut.configuration.jdbc.tomcat
 
-import io.micronaut.spring.tx.datasource.SpringDataSourceResolver
+import io.micronaut.jdbc.spring.SpringDataSourceResolver
 import org.apache.tomcat.jdbc.pool.DataSource
 import org.springframework.jdbc.datasource.TransactionAwareDataSourceProxy
 import spock.lang.Specification

--- a/jdbc/build.gradle
+++ b/jdbc/build.gradle
@@ -1,3 +1,8 @@
 dependencies {
+    compileOnly project(":inject-java")
+    compileOnly project(":spring")
+    compileOnly dependencyModuleVersion("spring", "spring-jdbc")
+    compileOnly dependencyModuleVersion("spring", "spring-tx")
+
     compile project(":inject")
 }

--- a/jdbc/src/main/java/io/micronaut/jdbc/spring/DataSourceTransactionManagerFactory.java
+++ b/jdbc/src/main/java/io/micronaut/jdbc/spring/DataSourceTransactionManagerFactory.java
@@ -14,13 +14,14 @@
  * limitations under the License.
  */
 
-package io.micronaut.spring.tx.datasource;
+package io.micronaut.jdbc.spring;
 
 import io.micronaut.context.annotation.EachBean;
 import io.micronaut.context.annotation.Factory;
 import io.micronaut.context.annotation.Requires;
 import io.micronaut.context.event.BeanCreatedEvent;
 import io.micronaut.context.event.BeanCreatedEventListener;
+import io.micronaut.core.annotation.Internal;
 import org.springframework.jdbc.datasource.DataSourceTransactionManager;
 import org.springframework.jdbc.datasource.TransactionAwareDataSourceProxy;
 
@@ -36,6 +37,7 @@ import javax.sql.DataSource;
 @Factory
 @Requires(classes = DataSourceTransactionManager.class)
 @Requires(condition = HibernatePresenceCondition.class)
+@Internal
 public class DataSourceTransactionManagerFactory {
 
     /**

--- a/jdbc/src/main/java/io/micronaut/jdbc/spring/HibernatePresenceCondition.java
+++ b/jdbc/src/main/java/io/micronaut/jdbc/spring/HibernatePresenceCondition.java
@@ -14,11 +14,12 @@
  * limitations under the License.
  */
 
-package io.micronaut.spring.tx.datasource;
+package io.micronaut.jdbc.spring;
 
 import io.micronaut.context.BeanContext;
 import io.micronaut.context.condition.Condition;
 import io.micronaut.context.condition.ConditionContext;
+import io.micronaut.core.annotation.Internal;
 import io.micronaut.core.reflect.ClassUtils;
 
 /**
@@ -27,6 +28,7 @@ import io.micronaut.core.reflect.ClassUtils;
  * @author graemerocher
  * @since 1.0
  */
+@Internal
 public final class HibernatePresenceCondition implements Condition {
     @Override
     public boolean matches(ConditionContext context) {

--- a/jdbc/src/main/java/io/micronaut/jdbc/spring/SpringDataSourceResolver.java
+++ b/jdbc/src/main/java/io/micronaut/jdbc/spring/SpringDataSourceResolver.java
@@ -14,8 +14,9 @@
  * limitations under the License.
  */
 
-package io.micronaut.spring.tx.datasource;
+package io.micronaut.jdbc.spring;
 
+import io.micronaut.core.annotation.Internal;
 import io.micronaut.jdbc.DataSourceResolver;
 import org.springframework.jdbc.datasource.DelegatingDataSource;
 
@@ -29,6 +30,7 @@ import javax.sql.DataSource;
  * @since 1.0
  */
 @Singleton
+@Internal
 public final class SpringDataSourceResolver implements DataSourceResolver {
 
     @Override

--- a/jdbc/src/main/java/io/micronaut/jdbc/spring/package-info.java
+++ b/jdbc/src/main/java/io/micronaut/jdbc/spring/package-info.java
@@ -15,9 +15,9 @@
  */
 
 /**
- * Classes for configuring transaction management for data sources.
+ * Classes for configuring transaction management via Spring for data sources.
  *
  * @author graemerocher
  * @since 1.0
  */
-package io.micronaut.spring.tx.datasource;
+package io.micronaut.jdbc.spring;

--- a/spring/build.gradle
+++ b/spring/build.gradle
@@ -5,7 +5,6 @@ dependencies {
     compile project(":inject")
     compileOnly project(":inject-java")
     compileOnly dependencyModuleVersion("spring", "spring-jdbc")
-    compileOnly project(":jdbc")
     compile project(":aop")
 
     testImplementation project(':runtime')

